src/pip/_internal/self_outdated_check.py

```diff
@@ -96,6 +100,102 @@ def was_installed_by_pip(pkg: str) -> bool:
     return dist is not None and "pip" == dist.installer
 
 
+def get_py_executable() -> str:
+    """Get path to launch a Python executable.
+
+    First test if python/python3/pythonX.Y on PATH matches the current
+    interpreter, and use that if possible. Then try to get the correct
+    pylauncher command to launch a process of the current python
+    version, fallback to sys.executable
+    """
+
+    if not sys.executable:
+        # docs (python 3.10) says that sys.executable can be can be None or an
+        # empty string if this value cannot be determined, although this is
+        # very rare. In this case, there is nothing much we can do
+        return "python3"
```

Comment:
This would be wrong on Windows, where there's never a `python3` executable. I'm not sure giving the wrong answer is better than explicitly failing to give any answer.

Reply:
I'm inclined to fail in this case.

Reply:
I was inclined to spit an error too, but I did not want the error to be too disruptive. But I see, giving a wrong suggestion does not help much either, this can be changed to put another warning instead (in place of the upgrade suggestion warning)